DaS_Programs/Singly linked list/Singly linked list/main.c

```diff
@@ -356,14 +362,32 @@ struct node *delete_before(struct node *start)
         
         preptr=start;
         ptr=start->next;
-        
-        while (ptr->next->data!=val)
+        if(preptr->data==val)
+		{
+			printf("\nThis is the first node. Nothing to delete\n");
+			return start;
+		}
+		if(ptr->data==val)
```

Comment:
Changes: This case is not needed, because the while loop takes care of this situation

Reply:
on my compiler. it didnt work for some reason.  besides the start value will change, right? so to combat that problem of new start value as well as due to above reason i added it

Reply:
Didnt get you boy. Can you run the code and put the result here please?

Reply:
haha 🤣  i'll resolve it, everything looks great